src/backend/Level0/CHIPBackendLevel0.cc

```diff
@@ -246,6 +246,10 @@ CHIPEventLevel0::~CHIPEventLevel0() {
     // '~CHIPEventLevel0' has a non-throwing exception specification
     assert(Status == ZE_RESULT_SUCCESS);
   }
+  if (EventPoolHandle_) {
+    auto Status = zeEventPoolDestroy(EventPoolHandle_);
+    assert(Status == ZE_RESULT_SUCCESS);
+  }
```

Comment:
What if there are other events still running that are part of this pool?

Reply:
I did not see user requested events are kept under same pool. Every time RT creates new event pool see the below code.
![image](https://github.com/CHIP-SPV/chip-spv/assets/52556743/4af757fb-3973-4e30-ada1-4496222ab8bb)

Reply:
Either we should not create pool every time while creating new events or we need to delete it while deleting the event.